### PR TITLE
doc: Path encoding in uv_fs_XX is UTF-8[Windows]

### PR DESCRIPTION
**Pull request by sainihitesh**
#1554 
Modified the documentation to mention that uv_fs_XXX functions use utf-8 encoding on Windows.
Native unicode encoding on Windows is UTF-16

### CODE REVIEW

File: docs/src/fs.rst [hunk 1]
```diff
@@ -175,6 +175,7 @@ API
         On Windows libuv uses `CreateFileW` and thus the file is always opened
         in binary mode. Because of this the O_BINARY and O_TEXT flags are not
         supported.
+        On Windows `uv_fs_XXX` functions use utf-8 encoding.
```
Comment (cjihrig) [inline docs/src/fs.rst]: Right now you have this in the documentation for `uv_fs_open()`. I don't think that's the best place for it. The first paragraph of the file might be a better place.

Comment (sainihitesh) [reply docs/src/fs.rst]: Ok @cjihrig.I will do the changes accordingly and make a pull request.

**Comment by sainihitesh** [comment]
Hi @cjihrig,
Added the Note to the first paragraph of the file.

File: docs/src/fs.rst (hunk 1 above)
Comment (sainihitesh) [reply docs/src/fs.rst]: Hi @cjihrig,
Added the Note to the first paragraph of the file.

File: docs/src/fs.rst [hunk 2]
```diff
@@ -12,6 +12,8 @@ otherwise it will be performed asynchronously.
 All file operations are run on the threadpool. See :ref:`threadpool` for information
 on the threadpool size.
 
+.. note::
+     On Windows `uv_fs_XXX` functions use utf-8 encoding.
```
Comment (bzoz) [inline docs/src/fs.rst]: I don't really like "`uv_fs_XXX`". I would prefer "`uv_fs_*`" or "all Libuv filesystem functions"

Comment (saghul) [reply docs/src/fs.rst]: Agreed.

File: docs/src/fs.rst (hunk 1 above)
Comment (sainihitesh) [reply docs/src/fs.rst]: Thanks @bzoz , @cjihrig for suggestions. I've made the required changes and pushed the commit.

**Comment by sainihitesh** [comment]
Thanks @bzoz , @cjihrig and @saghul  for suggestions. I've made the required changes and pushed the commit.

File: docs/src/fs.rst [hunk 3]
```diff
@@ -174,7 +176,7 @@ API
     .. note::
         On Windows libuv uses `CreateFileW` and thus the file is always opened
         in binary mode. Because of this the O_BINARY and O_TEXT flags are not
-        supported.
+        supported.
```
Comment (bzoz) [inline docs/src/fs.rst]: Extra white space at the end of the line. Other than that, this LGTM

**Comment by santigimeno** [comment]
Landed in https://github.com/libuv/libuv/commit/4772d44087a543c662864fd58b0a004d30618cc5. Thanks!

**Pull request closed without merging**